frontend/talentsearch/resources/js/components/workLocationPreferenceForm/WorkLocationPreferenceForm.tsx

```diff
@@ -0,0 +1,174 @@
+        );
+      });
+  };
+  const workRegionsForFormatedWorkPreferenceForm = defineMessages({
+    [WorkRegion.Telework]: {
+      defaultMessage: "Virtual Work from home, anywhere in Canada.",
+      description: "The work region of Canada described as Telework.",
+    },
+    [WorkRegion.NationalCapital]: {
+      defaultMessage: "National Capital Region: Ottawa, ON and Gatineau, QC.",
+      description: "The work region of Canada described as National Capital.",
+    },
+    [WorkRegion.Atlantic]: {
+      defaultMessage:
+        "Atlantic Region: New Brunswick, Newfoundland and Labrador, Nova Scotia and Prince Edward Island.",
+      description: "The work region of Canada described as Atlantic.",
+    },
+    [WorkRegion.Quebec]: {
+      defaultMessage: "Quebec Region: excluding Gatineau.",
+      description: "The work region of Canada described as Quebec.",
+    },
+    [WorkRegion.Ontario]: {
+      defaultMessage: "Ontario Region: excluding Ottawa.",
+      description: "The work region of Canada described as Ontario.",
+    },
+    [WorkRegion.Prairie]: {
+      defaultMessage: "Prairie Region: Manitoba, Saskatchewan, Alberta.",
+      description: "The work region of Canada described as Prairie.",
+    },
+    [WorkRegion.BritishColumbia]: {
+      defaultMessage: "British Columbia Region",
+      description: "The work region of Canada described as British Columbia.",
+    },
+    [WorkRegion.North]: {
+      defaultMessage: "North Region: Yukon, Northwest Territories and Nunavut.",
+      description: "The work region of Canada described as North.",
+    },
+  });
+
+  const getWorkPreferenceregion = (
+    workRegionId: string | number,
+  ): MessageDescriptor =>
+    getOrThrowError(
+      workRegionsForFormatedWorkPreferenceForm,
+      workRegionId,
+      `Invalid Work Region '${workRegionId}'`,
+    );
+
+  return (
+    <ProfileFormWrapper
+      description="Indicate all locations where you are willing to work, including your current location (if you are interested in working there)."
+      title="Work location"
+      crumbs={[
+        {
+          title: intl.formatMessage({
+            defaultMessage: "Work Location Preference",
+            description:
+              "Display Text for the current page in Work Location Preference Form Page",
+          }),
+        },
+      ]}
+    >
+      <BasicForm onSubmit={onSubmit}>
+        <div>
+          <div data-h2-flex-item="b(1of1)" data-h2-padding="b(top, m)">
+            <div data-h2-padding="b(right, l)" data-testid="workLocation">
+              <Checklist
+                idPrefix="work-location"
+                legend={intl.formatMessage({
+                  defaultMessage: "Work location",
+                  description:
+                    "Legend for optional work preferences check list in work preferences form",
+                })}
+                name="workLocations"
+                items={enumToOptions(WorkRegion).map(({ value }) => ({
```

Comment:
I think adding bold text here is more chellenging because its coming from localizedConstants

Reply:
Good point.  We would need to add the markup to the localizedConstant.  I still recommend adding the note on the issue.

Reply:
@petertgiles Who needs to do that issue?  I think its only small changes right?

Reply:
Nice, that's looking better!  I'll recommend one small change to your approach.  Maybe change `getworkRegionsDetailed` to `getWorkRegionDescriptions` and just include the descriptions.  This means you won't have to include the _bold_ markup in the string and will allow us to format it differently if needed.  Then your label would become something like:
```
   label: <bold>intl.formatMessage(getWorkRegions(value)) + ': '<strong> +
                 intl.formatMessage(getWorkRegionDescriptions(value))
```
Or maybe this will run into the same problem as before where you can't pass a ReactElement to a string prop?

Reply:
Ok. I will try that and let you know.

Reply:
@petertgiles ![image](https://user-images.githubusercontent.com/95497896/161272040-6de2731d-0ef1-495e-a0d4-56d5584c470f.png)
I am getting this error. Any sytax error you could find?